File: frontend/src/types/alert.ts
export interface AlertResponse {
  id: string;
  product_id: string;
  email: string;
  target_price: number;
  active: boolean;
  created_at: string;
}

export interface AlertCreateRequest {
  product_id: string;
  email: string;
  target_price: number;
}

export interface AlertUpdateRequest {
  target_price?: number;
  active?: boolean;
}

File: backend/app/schemas/alert.py
import uuid
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel


class AlertCreateRequest(BaseModel):
    product_id: uuid.UUID
    target_price: Decimal


class AlertUpdateRequest(BaseModel):
    target_price: Decimal | None = None
    active: bool | None = None


class AlertResponse(BaseModel):
    id: uuid.UUID
    product_id: uuid.UUID
    email: str
    target_price: float
    active: bool
    created_at: datetime

    model_config = {"from_attributes": True}
